unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiData.java

```diff
@@ -45,740 +45,740 @@
-	    .appendCodePoint(0x1F478)
-	    .appendCodePoint(0x1F934)
-	    .appendCodePoint(0x1F483)
-	    .appendCodePoint(0x1F57A)
+        .appendCodePoint(0x1F385)
+        .appendCodePoint(0x1F936)
+        .appendCodePoint(0x1F478)
+        .appendCodePoint(0x1F934)
+        .appendCodePoint(0x1F483)
+        .appendCodePoint(0x1F57A)
 
-	    .appendCodePoint(0x1F470)
-	    .appendCodePoint(0x1F935)
+        .appendCodePoint(0x1F470)
+        .appendCodePoint(0x1F935)
 
-	    .appendCodePoint(0x1F930)
-	    .appendCodePoint(0x1F931)
-	    .appendCodePoint(0x1F9D5)
+        .appendCodePoint(0x1F930)
+        .appendCodePoint(0x1F931)
+        .appendCodePoint(0x1F9D5)
 
 
-	    .appendCodePoint(0x1F574)
-	    .appendCodePoint(0x1F472)
-	    .toString();
+        .appendCodePoint(0x1F574)
+        .appendCodePoint(0x1F472)
+        .toString();
 
     public static final UnicodeSet hairBase = new UnicodeSet(
-	    "[👶 👮 👲 👳 👸 🕵 👼 💆 💇 👰 🙍 🙎 🙅 🙆 💁 🙋 🙇 " + "🚶 🏃 💃 🚣  🏄 🏊 ⛹ 🏋 " + "👦 👧  👩 👴 👵  "
-		    + "\\U0001F935 \\U0001F926 \\U0001F937 \\U0001F938 \\U0001F93B \\U0001F93C \\U0001F93D \\U0001F93E"
-		    + "\\U0001F934 \\U0001F936 \\U0001F57A \\U0001F930]").freeze();
+        "[👶 👮 👲 👳 👸 🕵 👼 💆 💇 👰 🙍 🙎 🙅 🙆 💁 🙋 🙇 " + "🚶 🏃 💃 🚣  🏄 🏊 ⛹ 🏋 " + "👦 👧  👩 👴 👵  "
+            + "\\U0001F935 \\U0001F926 \\U0001F937 \\U0001F938 \\U0001F93B \\U0001F93C \\U0001F93D \\U0001F93E"
+            + "\\U0001F934 \\U0001F936 \\U0001F57A \\U0001F930]").freeze();
     public static final UnicodeSet directionBase = new UnicodeSet(
-	    "[😘 🚶 🏃 ✌ ✋ 👋-👍 👏 💪 👀 🤘 💨 ✈ 🎷 🎺 🔨 ⛏ 🗡 🔫 🚬 "
-		    + "\\U0001F93A \\U0001F93D \\U0001F93E \\U0001F946]").freeze();
+        "[😘 🚶 🏃 ✌ ✋ 👋-👍 👏 💪 👀 🤘 💨 ✈ 🎷 🎺 🔨 ⛏ 🗡 🔫 🚬 "
+            + "\\U0001F93A \\U0001F93D \\U0001F93E \\U0001F946]").freeze();
     private static final boolean SHOW = false;
 
     /**@deprecated Replace by the CountEmoji.Category*/
     public enum ZwjType {
-	roleWithHair, roleWithObject, roleWithSign, gestures, activity, family, other, na;
-	public static ZwjType getType(String s) {
-	    if (!s.contains(Emoji.JOINER_STRING)) {
-		return na;
-	    }
-	    int[] cps = CharSequences.codePoints(s);
-	    ZwjType zwjType = ZwjType.other;
-	    if (Emoji.HAIR_PIECES.containsSome(s)) {
-		zwjType = roleWithHair;
-	    } else if (Emoji.FAMILY_MARKERS.contains(cps[cps.length - 1])) { // last character is in boy..woman
-		zwjType = family;
-	    } else if (Emoji.ACTIVITY_MARKER.containsSome(s)) {
-		zwjType = activity;
-	    } else if (Emoji.ROLE_MARKER.containsSome(s)) { //  || Emoji.FAMILY_MARKERS.containsSome(s)
-		zwjType = Emoji.GENDER_MARKERS.containsSome(s) ? roleWithSign : roleWithObject;
-	    } else if (Emoji.GENDER_MARKERS.containsSome(s)) {
-		zwjType = gestures;
-	    }
-	    return zwjType;
-	}
+    roleWithHair, roleWithObject, roleWithSign, gestures, activity, family, other, na;
+    public static ZwjType getType(String s) {
+        if (!s.contains(Emoji.JOINER_STRING)) {
+        return na;
+        }
```

Comment:
This file also wanted 8sp/tab.

Reply:
Fixed in https://github.com/unicode-org/unicodetools/pull/178/commits/7c5c10bdc0232617b772ea6dd0e1e9389a3a9cfc